app/policies/user_policy.rb

```diff
@@ -0,0 +1,24 @@
+class UserPolicy < ApplicationPolicy
+  attr_reader :user, :record
+
+  def initialize(user, record)
+    @user = user
+    @record = record
+  end
+
+  def edit?
+    update?
+  end
+
+  def update?
+    user == record || user.sys_admin_role? || user.admin_role?
```

Comment:
should users bc able to edit their own roles tho? maybe dropdown is only their permission level and lower?

Reply:
yeah I think the controller layer is the right place to set that